adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -119,6 +118,480 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   def transform(tFn: RDD[T] => RDD[T]): U = {
     replaceRdd(tFn(rdd))
   }
+  val self = this
+
+  /**
+   *
+   */
+  private[rdd] object SortedTrait extends Serializable {
+    var isSorted: Boolean = false
```

Comment:
This is a bit more challenging than initially thought. I will consider a case class, but it is not trivial to simply make this immutable and keep as a singleton object. I will work on this.